lib/bakeware/cpio.ex

```diff
@@ -0,0 +1,101 @@
+defmodule Bakeware.CPIO do
+  alias Bakeware.Assembler
+
+  @cpio_magic "070701"
+
+  def build(%Assembler{} = assembler) do
+    File.open(assembler.cpio, [:write, :append], fn cpio_fd ->
+      _ = append_files(assembler.rel_path, cpio_fd)
+      _ = append_trailer(cpio_fd)
+    end)
+
+    _ = maybe_compress(assembler)
+
+    assembler
+  end
+
+  defp append_files(release_path, cpio_fd) do
+    for path <- Path.wildcard("#{release_path}/**/*"), path != release_path do
+      file = file_details(path, release_path)
+
+      # write CPIO header
+      IO.binwrite(cpio_fd, build_header(file.mode, file.size, file.relative_path))
+
+      maybe_write_file_contents(file, cpio_fd)
+    end
+  end
+
+  defp append_trailer(fd) do
+    IO.binwrite(fd, build_header(0, 0, "TRAILER!!!"))
+  end
+
+  defp build_header(mode, size, relative) do
+    namesize = byte_size(relative)
+
+    [
+      @cpio_magic,
+      zero(),
+      pad_hex(mode),
+      zero(),
+      zero(),
+      zero(),
+      zero(),
+      pad_hex(size),
+      zero(),
+      zero(),
+      zero(),
+      zero(),
+      pad_hex(namesize + 1),
+      zero(),
+      relative,
+      # Null terminator on path
+      0,
+      pad_to_4(110 + namesize + 1)
+    ]
+  end
+
+  defp file_details(path, release_path) do
+    File.stat!(path)
+    |> Map.from_struct()
+    |> Map.put(:path, path)
+    |> Map.put(:relative_path, Path.relative_to(path, release_path))
+  end
+
+  defp maybe_compress(%{compress?: true} = assembler) do
+    out = assembler.cpio <> ".z"
+    :os.cmd('zstd -15 #{assembler.cpio} -o #{out} --rm')
```

Comment:
```suggestion
    {_, 0} = System.cmd("zstd", "-15", assembler.cpio, "-o", out, "--rm")
```
That should handle paths with spaces. Another concern would be to print a nice message if there's an error, but I'm not sure what all can go wrong.

Also, and I know this doesn't matter since it's a temp file, but the normal file extension to zstd files is `.zst` and not `.z`.

Reply:
Done